### PR TITLE
Extra keyed_set to support compound keys

### PR DESCRIPTION
**Pull request by charleso**

### CODE REVIEW

File: ivory-core/src/main/thrift/core.thrift [hunk 1]
```diff
@@ -142,6 +142,7 @@ enum ThriftDictionaryMode {
 union ThriftDictionaryModeV2 {
     1: ThriftDictionaryMode mode,
     2: string keyedSet
+    3: list<string> keyedSetMulti
```
Comment (markhibberd) [inline ivory-core/src/main/thrift/core.thrift]: Mmm... I think we should possibly bumping the metadata version with these changes - really we want things to not understand this to fail right? Need to think about it a bit more...

File: ivory-core/src/test/scala/com/ambiata/ivory/core/gen/GenDictionary.scala [hunk 2]
```diff
@@ -10,7 +10,10 @@ import scalaz.scalacheck.ScalaCheckBinding._
 object GenDictionary {
   def mode: Gen[Mode] =
     // By generating a KeyedSet we force the encoding to be a struct, so let's generate less of them
-    Gen.frequency(9 -> modeImplemented, 1 -> GenString.sensible.map(Mode.KeyedSet))
+    Gen.frequency(
+      9 -> modeImplemented,
+      1 -> GenPlus.listOfSized(1, 3, GenString.sensible).map(Mode.KeyedSet)
+    )
 
   // Remove when KeyedSet is supported fully
```
Comment (markhibberd) [inline ivory-core/src/test/scala/com/ambiata/ivory/core/gen/GenDictionary.scala]: Can this comment be removed now?

File: ivory-core/src/main/thrift/core.thrift (hunk 1 above)
Comment (charleso) [inline ivory-core/src/main/thrift/core.thrift]: > really we want things to not understand this to fail right?

Yep, good point.

**Comment by markhibberd** [comment]
Looks pretty good. Good test for implementation that we can make this sort of change without too much hassle.

File: ivory-core/src/main/thrift/core.thrift [hunk 3]
```diff
@@ -1,5 +1,13 @@
 namespace java com.ambiata.ivory.core.thrift
 
+/**
+ * ====================================================================================================
+ *
+ * NOTE: Please update the Ivory metadata version for _any_ non-forward compatible change to this file.
```
Comment (charleso) [inline ivory-core/src/main/thrift/core.thrift]: Warning added

File: ivory-core/src/test/scala/com/ambiata/ivory/core/gen/GenDictionary.scala (hunk 2 above)
Comment (charleso) [inline ivory-core/src/test/scala/com/ambiata/ivory/core/gen/GenDictionary.scala]: > Can this comment be removed now?

The function should have been removed before in the previous PR. Done now.

**Pull request merged**